packages/core/lib/commands/opcode/run.js

```diff
@@ -21,14 +21,22 @@ module.exports = async function (options) {
     );
   }
 
-  let bytecode = Contract.deployedBytecode;
-  let numInstructions = Contract.deployedSourceMap.split(";").length;
+  const bytecode = options.creation
+    ? Contract.bytecode
+    : Contract.deployedBytecode;
+  const sourceMap = Compilations.Utils.simpleShimSourceMap(
+    options.creation ? Contract.sourceMap : Contract.deployedSourceMap
+  ); //the shim is in case of an old-style Vyper source map
```

Comment:
Is this shim useful elsewhere? If so, perhaps it should be the responsibility of either `@truffle/contract` or `@truffle/resolver` to conditionally apply it when applicable?

Reply:
I mean, we already use this shim elsewhere!  This PR didn't create the shim, it just used the one I'd already written.

If we were going to do something like what you suggest, I imagine it would be in `@truffle/config`.  I don't think that would be a good idea though, because I think it could potentially limit us in the future.  Way way down on my list of priorities somewhere is issue #3830, which that sort of auto-shimming would make impossible.  Granted, I am pretty skeptical I'll ever *actually* bother to implement a solution for #3830 (I'm surprised it got tagged priority 6 instead of 7!), but the potential for other issues of that sort in the future make me wary of doing that sort of thing.

Reply:
Hopefully in that case the shim detects if it's already been applied and bails early, and/or the lifecycle of a shimmed contract is always quite short when this is in use?

I just tend to view mutations like this as difficult-to-untangle consistency issues lying in wait.

Reply:
Ah but I'm a dolt - you're not mutating the contract instance - it's returning the shimmed `sourceMap`. Nevermind me.